src/email/client.cr

```diff
@@ -196,7 +203,13 @@ class EMail::Client
           log_error("TLS is disabled because `-D without_openssl` was passed at compile time")
           false
         {% else %}
-          tls_socket = OpenSSL::SSL::Socket::Client.new(@socket.as(TCPSocket), sync_close: true, hostname: @host)
+          tls_context = OpenSSL::SSL::Context::Client.new
+          if !@openssl_verify_mode.nil? && OPENSSL_VERIFICATION_MODES.keys.index(@openssl_verify_mode.to_s)
```

Comment:
I think the condition of this `if` will be false when the `openssl_verify_mode` option is not set, and the error message will be logged.

Is the `openssl_verify_mode` option` option required?